torchvision/datasets/gtsrb.py

```diff
@@ -0,0 +1,114 @@
+import csv
+import os
+from typing import Any, Callable, Optional, Tuple
+
+import PIL
+
+from .folder import make_dataset
+from .utils import download_and_extract_archive
+from .vision import VisionDataset
+
+
+class GTSRB(VisionDataset):
+    """`German Traffic Sign Recognition Benchmark (GTSRB) <https://benchmark.ini.rub.de/>`_ Dataset.
+
+    Args:
+        root (string): Root directory of the dataset.
+        train (bool, optional): If True, creates dataset from training set, otherwise
+            creates from test set.
+        transform (callable, optional): A function/transform that  takes in an PIL image and returns a transformed
+            version. E.g, ``transforms.RandomCrop``.
+        target_transform (callable, optional): A function/transform that takes in the target and transforms it.
+        download (bool, optional): If True, downloads the dataset from the internet and
+            puts it in root directory. If dataset is already downloaded, it is not
+            downloaded again.
+    """
+
+    # Ground Truth for the test set
+    gt_url = "https://sid.erda.dk/public/archives/daaeac0d7ce1152aea9b61d9f1e19370/GTSRB_Final_Test_GT.zip"
+    gt_csv = "GT-final_test.csv"
+    gt_md5 = "fe31e9c9270bbcd7b84b7f21a9d9d9e5"
+
+    # URLs for the test and train set
+    urls = (
+        "https://sid.erda.dk/public/archives/daaeac0d7ce1152aea9b61d9f1e19370/GTSRB_Final_Test_Images.zip",
+        "https://sid.erda.dk/public/archives/daaeac0d7ce1152aea9b61d9f1e19370/GTSRB-Training_fixed.zip",
+    )
+
+    md5 = ("c7e4e6327067d32654124b0fe9e82185", "513f3c79a4c5141765e10e952eaa2478")
+
+    extension = (".ppm",)
+
+    def __init__(
+        self,
+        root: str,
+        train: bool = True,
+        transform: Optional[Callable] = None,
+        target_transform: Optional[Callable] = None,
+        download: bool = False,
+    ) -> None:
+
+        super().__init__(root, transform=transform, target_transform=target_transform)
+
+        self.root = os.path.expanduser(root)
+
+        self.train = train
+        self.url = self.urls[self.train]
+        self.md5 = self.md5[self.train]
+
+        if download:
+            self.download()
+
+        if not self._check_exists():
+            raise RuntimeError("Dataset not found. You can use download=True to download it")
+
+        if train:
+            samples = make_dataset(self._target_folder, extensions=self.extension)
+        else:
+            with open(os.path.join(self._base_folder, self.gt_csv)) as csv_file:
+                data = list(csv.reader(csv_file, delimiter=";", skipinitialspace=True))
+            data = data[1:]
+            samples = [(os.path.join(self._target_folder, row[0]), int(row[-1])) for row in data]
```

Comment:
This is a little hard to read. I've added a suggestion that does exactly the same and is (hopefully) easier to read

```suggestion
                samples = [
                    (os.path.join(self._target_folder, row["Filename"]), int(row["ClassId"]))
                    for row in csv.DictReader(csv_file, delimiter=";", skipinitialspace=True)
                ]
```

Reply:
Thanks for the suggestion, I think it is definitely more easier to read. I will update the code.